libnmstate/nm/nmclient.py

```diff
@@ -220,3 +229,57 @@ def _register_first_action(self):
     def _execute_action_once(self, _):
         self.execute_next_action()
         return False
+
+
+class _NmClient:
+    def __init__(self):
+        self._client = NM.Client.new(None)
+        logging.debug("NM.Client created")
+        self._mainloop = None
+
+    @property
+    def client(self):
+        return self._client
+
+    def delete_client(self):
+        if not self.client:
+            return
+        try:
+            ref_count = sys.getrefcount(self.client)
+            if ref_count != 2:
+                # The getrefcount() hold 1 reference and self.client itself
+                # hold 1 reference.
+                raise error.NmstateInternalError(
+                    "_NmClient.delete_client() cannot release resources "
+                    "of NM.Client duo to other unreleased use."
+                )
+            if NM.Client.get_version(self.client) >= StrictVersion("1.21.0"):
```

Comment:
IMHO it would be nicer to just check whether `get_context_busy_watcher()` exists/accessing it raises an `AttributeError` instead of guessing it from the version.